main/utils.py

```diff
@@ -261,6 +261,11 @@ def get_field_names(model):
     ]
 
 
+def filter_dict_by_key_set(dict_to_filter, key_set):
```

Comment:
Can we add a test for this?

Reply:
LOL, accidentally added this from another branch. Test added anyway